internal/packetcapture/packetcapture.go

```diff
@@ -53,34 +54,38 @@ func (pc *PacketCapture) RegisterReceiver(receiver PacketReceiver) {
 }
 
 func (pc *PacketCapture) Start() error {
-	inactive, err := pcap.NewInactiveHandle(captureDevice)
-	if err != nil {
-		return err
-	}
-	defer inactive.CleanUp()
-
-	// Force packets to be sent immediately to ensure we don't miss any.
-	if err := inactive.SetImmediateMode(true); err != nil {
-		return err
-	}
-
-	handle, err := inactive.Activate()
+	// Use the pcapgo library for capturing traffic as it is the most reliable.
+	// afpacket is the fastest but segfaults: https://github.com/google/gopacket/issues/717
+	// pcap will block during Cancel(): https://github.com/google/gopacket/issues/890
+	handle, err := pcapgo.NewEthernetHandle(pc.netInterface)
 	if err != nil {
 		return err
 	}
 	pc.handle = handle
-	packetSource := gopacket.NewPacketSource(pc.handle, pc.handle.LinkType())
-	go func(packets chan gopacket.Packet, done chan bool) {
-		for packet := range packets {
-			pc.handlePacket(packet)
+	packetSource := gopacket.NewPacketSource(pc.handle, layers.LinkTypeEthernet)
+	go func(packets chan gopacket.Packet, done chan bool, stop chan bool) {
+		defer func() { done <- true }()
+		for {
+			select {
+			case packet, ok := <-packets:
+				if ok {
+					pc.handlePacket(packet)
+				} else {
+					return
```

Comment:
Is there some error you noticed which led you to add this? Should we log an error here?

Reply:
Yes and no.

1. I was always bothered by the implicit `pc.handle.Close()` method for exiting the go routine, this approach is more explicit.
2. Before I switched to `pcapgo` the `pc.handle.Close()` would block, I tried this change to see if it would fix the issue.

Reply:
There isn't really an error to log here either.

Reply:
Ack.